test/extensions/transport_sockets/tls/ocsp/ocsp_test.cc

```diff
@@ -189,6 +189,27 @@ TEST_F(Asn1OcspUtilityTest, ParseMethodWrongTagTest) {
   expectThrowOnWrongTag(Asn1OcspUtility::parseResponseStatus);
 }
 
+TEST_F(Asn1OcspUtilityTest, ParseResponseDataUnsupportedVersionTest) {
+  std::vector<uint8_t> data = {
+      // SEQUENCE
+      0x30,
+      7,
+      // invalid version 1
+      0xa0,
+      0x02,
+      1,
+      0,
+      // Responder ID tag 1
+      1,
+      1,
+      0,
+  };
```

Comment:
This was confusing me because the integer expressed here (0x02 1 0) has the value 0, which should pass the test. The reason is that the field is:
```
     version              [0] EXPLICIT Version DEFAULT v1
```
and `EXPLICIT` means that the integer is wrapped in an outer encoding. I think the correct representation of a version with integer value 1 is therefore:

```suggestion
  std::vector<uint8_t> data = {
      // SEQUENCE
      0x30,
      8,
      // invalid version 1
      0xa0,
      0x03, // bitstring
      0x02, // integer
      1,
      1,
      // Responder ID tag 1
      1,
      1,
      0,
  };
```

That also means that the way the value is being extracted is incorrect. See my note in oscp.cc.

Reply:
Thanks for input, I will update.